### PR TITLE
Fix topic name null check and remove redundant stuff

### PR DESCRIPTION
**Pull request by tapishr**
## Summary by CodeRabbit

- **Refactor**
	- Improved the logic and error handling in the publication and subscription services.
	- Streamlined topic creation by removing unnecessary user-specific parameters.

- **Chores**
	- Cleaned up code by removing unused imports.

### CODE REVIEW

**Review by MuskanPaliwal — DISMISSED** [review]

File: pages/api/dpu/pubsub.ts [hunk 1]
```diff
@@ -18,30 +17,31 @@ const pubsubHandler = async (req: NextApiRequest, res: NextApiResponse) => { //
...
             res.status(500).json({"error": "Internal server error"});
             return;
         }
-        topicName = generated_topic;
-        saveTopicNameInUsersTable(jsonBody.user_id, topicName)
+        saveTopicNameInUsersTable(jsonBody.user_id, generated_topic)
         .then(() => { console.info("[pubsubHandler] Topic saved in db"); })
         .catch((error) => {
             console.error("[pubsubHandler] Unable to save topic name in db, ", error);
         })
```
[the remark this answers is not in the document]
Comment (avikalpg) [reply pages/api/dpu/pubsub.ts:37]: @tapishr, although I understand that saving in the database is not immediately required for returning this value, but can you please confirm that even if the topic name is not saved in the database, the user will be able to continue with their setup?

If not, then maybe we should `await` this function call.

**Review by avikalpg — DISMISSED** [review]
Everything looks good. I just have one minor comment.

File: pages/api/dpu/pubsub.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (MuskanPaliwal) [reply pages/api/dpu/pubsub.ts:37]: Agreed, this should be handled. Topic name should be saved in db only then we should proceed further

[the remark this answers is not in the document]
Comment (tapishr) [reply pages/api/dpu/pubsub.ts]: Done

**Review by MuskanPaliwal — APPROVED** [review]

**Review by avikalpg — APPROVED** [review]

**Pull request merged**